flow/src/main/scala/org/alephium/flow/core/BlockFlowGroupView.scala

```diff
@@ -134,19 +134,15 @@ object BlockFlowGroupView {
           case Some(_) =>
             Left(WorldState.expectedContractError)
           case None =>
-            val index = blockCaches.indexWhere(_.relatedOutputs.contains(outputRef))
-            if (index != -1) {
-              blockCaches(index).relatedOutputs.get(outputRef) match {
-                case Some(output: ContractOutput) =>
-                  Right(Some(output))
-                case Some(_) =>
-                  Left(WorldState.expectedContractError)
-                case None =>
-                  Right(None)
+            blockCaches.iterator
+              .collectFirst {
```

Comment:
Let's revert this change. Scala std collection's higher-order functions are generally slow.

Since we know why that line won't be covered, it's fine that we leave it as it is.

Reply:
Or we could introduce `collectFirst` for `AVector`